llvm/test/Transforms/LoopVectorize/dbg-outer-loop-vect.ll

```diff
@@ -15,8 +15,8 @@ define void @foo(ptr %h) !dbg !4 {
 ; CHECK-NEXT:    [[INDEX:%.*]] = phi i64 [ 0, [[VECTOR_PH]] ], [ [[INDEX_NEXT:%.*]], [[FOR_COND_CLEANUP32:%.*]] ]
 ; CHECK-NEXT:    br label [[FOR_COND5_PREHEADER1:%.*]], !dbg [[DBG21]]
 ; CHECK:       for.cond5.preheader1:
-; CHECK-NEXT:    [[VEC_PHI:%.*]] = phi <4 x i64> [ zeroinitializer, [[VECTOR_BODY]] ], [ [[TMP4:%.*]], [[FOR_COND5_PREHEADER1]] ], !dbg [[DBG21]]
-; CHECK-NEXT:    [[TMP0:%.*]] = getelementptr i32, ptr [[H]], <4 x i64> [[VEC_PHI]], !dbg [[DBG21]]
+; CHECK-NEXT:    [[VEC_PHI:%.*]] = phi <4 x i64> [ zeroinitializer, [[VECTOR_BODY]] ], [ [[TMP4:%.*]], [[FOR_COND5_PREHEADER1]] ]
```

Comment:
Why is the debug loc dropped here?

Reply:
Somehow there is no debug location information in the  original scalar `phi` and following `gep`. 
The previous debug information of the vector `phi`  seems from the previous `br lable [[FOR_COND5_PREHEADER1:%.*]], !dbg [[DBG21]]`, which might be incorrect.

Reply:
Ok, please split off the changes to VPWidenPHIRecipe then. Technically VPWidenIntrinsic/VPReduction changes are also independent, might be good to keep in mind for future PRs.

Reply:
That would be great to get split off, part of those changes are also tied up in #118662

Reply:
Split to #120338. Thanks.